core/src/main/java/ai/timefold/solver/core/config/solver/EnvironmentMode.java

```diff
@@ -106,22 +130,46 @@ public enum EnvironmentMode {
         this.asserted = asserted;
     }
 
+    public boolean isStepAssertOrMore() {
```

Comment:
Do we have unit tests for this?

Reply:
I had to adjust quite a lot of coverage for this.